internal/app/init/pkg/system/service_runner.go

```diff
@@ -0,0 +1,121 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package system
+
+import (
+	"context"
+	"log"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/talos-systems/talos/internal/app/init/pkg/system/runner"
+	"github.com/talos-systems/talos/pkg/userdata"
+)
+
+// ServiceState is enum of service run states
+type ServiceState int
+
+// ServiceState constants
+const (
+	StateInitialized ServiceState = iota
+	StatePrepare
+	StateWaiting
+	StateRunning
+	StateFinished
+	StateFailed
+)
+
+// ServiceEvent describes state change of the running service
+
+// ServiceRunner wraps the state of the service (running, stopped, ...)
+type ServiceRunner struct {
+	mu sync.Mutex
+
+	userData *userdata.UserData
+	service  Service
+	state    ServiceState
```

Comment:
not implemented yet, but ability to see service state (crashed, restarting, etc.), plus I plan to add "events []string" which would publish last N events for the service (similar to k8s)